### PR TITLE
Feature/home page

### PR DESCRIPTION
**Pull request by netashefer**

### CODE REVIEW

File: src/communication/dashboardCommunicator.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+                dashboardId,
+                username: "neta"
+            };
+            return requestProvider.post(this.getFullURL("dashboards/addPermissions"), { dashboardPermissions });
+        } catch (e) {
+            throw e;
+        }
+    }
+
+    getMyDashboards() {
```
Comment (YahelR) [inline src/communication/dashboardCommunicator.ts:29]: hahaha

File: src/components/DashboardPage/DashbaordPage.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import { FullDataInstanceInfo, Table, TableDictionary } from "../../types/data";
+import Dashboard from "../Dashboard/Dashboard";
+import ManagerPanel from "../ManagerPanel/ManagerPanel";
+
+const DashbaordPage = () => {
+    const [tableDictionary, setTableDictionary] = useState<TableDictionary>({});
+    const [fullDataInstanceInfo, setFullDataInstanceInfo] = useState<FullDataInstanceInfo>({});
+
+    const addDataInstanceTable = async (table: Table, info: any) => {
+        const dataInstanceId = uuidv4();
```
Comment (YahelR) [inline src/components/DashboardPage/DashbaordPage.tsx]: I am again asking if this is something that is supposed to be generated in the client, I feel like it should be generated in the server

File: src/components/LogoTitle/LogoTitle.tsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
+
+import { ReactComponent as Logo } from '../../icons/logo.svg';
+import './LogoTitle.scss';
+
+const LogoTitle = ({ size }: { size: 'sm' | 'lr'; }) => {
+    return (
+
```
Comment (YahelR) [inline src/components/LogoTitle/LogoTitle.tsx]: whiteline

File: src/components/HomePage/HomePage.tsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import CreateNewDashboard from "../CreateNewDashboard/CreateNewDashboard";
+import LogoTitle from "../LogoTitle/LogoTitle";
+import MyDashboards from "../MyDashboards/MyDashboards";
+import TopBar from "../TopBar/TopBar";
+import './HomePage.scss';
+
+const HomePage = () => {
+
```
Comment (YahelR) [inline src/components/HomePage/HomePage.tsx]: no whiteline

File: src/components/MyDashboards/DashboardItem/DashboardItem.tsx [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+type DashbaordItemProps = Dashboard;
+
+const DashbaordItem = ({ dashboardName, dashboardId }: DashbaordItemProps) => {
+
+    const deleteDashboard = () => {
+        dashboardCommunicator.deleteDashboard(dashboardId); // update recoil
+    };
+
+    return (
+
```
Comment (YahelR) [inline src/components/MyDashboards/DashboardItem/DashboardItem.tsx]: what is this whiteline

File: src/components/MyDashboards/DashboardItem/DashboardItem.tsx [hunk 6]
```diff
@@ -0,0 +1,26 @@
+import dashboardCommunicator from '../../../communication/dashboardCommunicator';
+import { ReactComponent as Trash } from '../../../icons/trash.svg';
+import { Dashboard } from '../../../types/entities';
+import './DashboardItem.scss';
+
+type DashbaordItemProps = Dashboard;
+
+const DashbaordItem = ({ dashboardName, dashboardId }: DashbaordItemProps) => {
+
```
Comment (YahelR) [inline src/components/MyDashboards/DashboardItem/DashboardItem.tsx:7]: whiteline

File: src/components/MyDashboards/DashboardItem/DashboardItem.tsx [hunk 7]
```diff
@@ -0,0 +1,26 @@
+import dashboardCommunicator from '../../../communication/dashboardCommunicator';
+import { ReactComponent as Trash } from '../../../icons/trash.svg';
+import { Dashboard } from '../../../types/entities';
+import './DashboardItem.scss';
+
+type DashbaordItemProps = Dashboard;
+
+const DashbaordItem = ({ dashboardName, dashboardId }: DashbaordItemProps) => {
```
Comment (YahelR) [inline src/components/MyDashboards/DashboardItem/DashboardItem.tsx]: Dashboard

File: src/components/MyDashboards/DashboardItem/DashboardItem.tsx [hunk 8]
```diff
@@ -0,0 +1,26 @@
+import dashboardCommunicator from '../../../communication/dashboardCommunicator';
+import { ReactComponent as Trash } from '../../../icons/trash.svg';
+import { Dashboard } from '../../../types/entities';
+import './DashboardItem.scss';
+
+type DashbaordItemProps = Dashboard;
```
Comment (YahelR) [inline src/components/MyDashboards/DashboardItem/DashboardItem.tsx]: typo

File: src/components/MyDashboards/MyDashboards.tsx [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+import './MyDashboards.scss';
+
+const MyDashboards = () => {
+    const [searchedValue, setSearchedValue] = useState("");
+    const [myDashboards, setMyDashboards] = useState<Dashboard[]>([]); // need recoil
+
+    useEffect(() => {
+        (async () => {
+            const dashabords = await dashboardCommunicator.getMyDashboards();
+            setMyDashboards(dashabords);
```
Comment (YahelR) [inline src/components/MyDashboards/MyDashboards.tsx]: you wrote dashboards in 3 different ways so far 🤠

File: src/components/MyDashboards/MyDashboards.tsx [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+        })();
+    }, []);
+
```
Comment (YahelR) [inline src/components/MyDashboards/MyDashboards.tsx:19]: whitelineeeeeeeee

File: src/components/MyDashboards/MyDashboards.tsx [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+
+    const setSearched = (event: ChangeEvent<HTMLInputElement>) => {
+        setSearchedValue(event.target.value);
+    };
+
+    return (
+
```
Comment (YahelR) [inline src/components/MyDashboards/MyDashboards.tsx]: no

File: src/components/MyDashboards/MyDashboards.tsx [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+            <div className="list">
+                {
+                    myDashboards?.
+                        filter(d => d.dashboardName?.includes(searchedValue))
+                        .map(d => <DashbaordItem
+                            dashboardId={d.dashboardId}
+                            dashboardName={d.dashboardName}
+                        />)
+                }
+
```
Comment (YahelR) [inline src/components/MyDashboards/MyDashboards.tsx]: :(

File: src/components/MyDashboards/MyDashboards.tsx [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+            </div>
+
```
Comment (YahelR) [inline src/components/MyDashboards/MyDashboards.tsx]: 2

File: src/components/MyDashboards/MyDashboards.tsx [hunk 14]
```diff
@@ -0,0 +1,58 @@
...
+        </div>
+
```
Comment (YahelR) [inline src/components/MyDashboards/MyDashboards.tsx]: 3

File: src/components/TopBar/TopBar.scss [hunk 15]
```diff
@@ -0,0 +1,9 @@
+.top-bar {
+    height: 10%;
+    background-image: linear-gradient(to right, #5462E1, #11756A);
+    min-height: 4em;
+    display: flex;
+    width: 100%;
+
```
Comment (YahelR) [inline src/components/TopBar/TopBar.scss]: 1

File: src/components/TopBar/TopBar.scss [hunk 16]
```diff
@@ -0,0 +1,9 @@
...
+
```
Comment (YahelR) [inline src/components/TopBar/TopBar.scss]: 2

File: src/components/TopBar/TopBar.tsx [hunk 17]
```diff
@@ -0,0 +1,20 @@
+
```
Comment (YahelR) [inline src/components/TopBar/TopBar.tsx:1]: 1

File: src/components/TopBar/TopBar.tsx [hunk 18]
```diff
@@ -0,0 +1,20 @@
...
+import LogoTitle from '../LogoTitle/LogoTitle';
+import './TopBar.scss';
+
```
Comment (YahelR) [inline src/components/TopBar/TopBar.tsx:4]: 2

File: src/components/DashboardPage/DashbaordPage.tsx (hunk 2 above)
Comment (netashefer) [reply src/components/DashboardPage/DashbaordPage.tsx]: its changed in the excel branch

File: src/components/MyDashboards/DashboardItem/DashboardItem.tsx (hunk 5 above)
Comment (netashefer) [reply src/components/MyDashboards/DashboardItem/DashboardItem.tsx]: copy paste for not writing from the beginning :)

File: src/components/MyDashboards/MyDashboards.tsx (hunk 9 above)
Comment (netashefer) [reply src/components/MyDashboards/MyDashboards.tsx]: חחחחחחחחחחחחחחח

**Pull request merged**